jxmapviewer2/src/main/java/org/jxmapviewer/JXMapViewer.java

```diff
@@ -298,9 +298,9 @@ private void drawOverlays(final int zoom, final Graphics g, final Rectangle view
 	}
 
 	@SuppressWarnings("unused")
-	private boolean isTileOnMap(int x, int y, Dimension mapSize)
+	private boolean isTileOffTheMap(int x, int y, Dimension mapSize)
 	{
-		return !isNegativeYAllowed && y < 0 || y >= mapSize.getHeight();
```

Comment:
This fixed the issue #45 for me (as in this case a background is properly rendered instead of loading image which is later not properly hidden) - if I overlooked something and this isn't a proper fix please simply discard it.